app/controllers/concerns/filter_parameters.rb

```diff
@@ -25,7 +19,33 @@ def deserialize_array_filter_param(param)
     params[param].split(",") if params.key?(param)
   end
 
+  def remove_previous_parameters(all_parameters)
+    all_parameters.except(*previous_parameters.keys)
+  end
+
+  def merge_previous_parameters(all_parameters)
+    previous_parameters.each do |key, value|
+      next if key =~ /prev_/ && value == "none"
+
+      all_parameters[key.split("_").last] = value
```

Comment:
```suggestion
      all_parameters[key.delete_prefix("prev_")] = value
```

To protect against a subtle gotcha in the future, esp with the params being refactored a little bit. Also better communicates intent to reader.